solver/src/solver/http_solver/model.rs

```diff
@@ -41,7 +42,7 @@ pub enum PoolType {
 #[derive(Debug, Serialize)]
 pub struct PoolModel {
     pub pool_type: PoolType,
-    pub reserves: HashMap<String, PoolTokenData>,
+    pub reserves: HashMap<H160, PoolTokenData>,
     pub fee: f64,
```

Comment:
I'm not sure we should use a 64-bit floating point number here because of precision issues.For the Uniswap fee example:
```rust
println!("{:.64}", 0.3); // 0.2999999999999999888977697537484345957636833190917968750000000000
println!("{:.64}", 0.003) // 0.0030000000000000000624500451351650553988292813301086425781250000
```

Maybe we should use arbitrary precision decimals (`BigDecimal`) so that it serializes as a string?

Reply:
This is a great idea, but the only problem is that 

```
the trait `Serialize` is not implemented for `BigDecimal`
```

Reply:
Note that there is a `serde` feature to add serialization support:

https://github.com/akubera/bigdecimal-rs/blob/fc797c302f4947e429cc331febc66fe3c7badc5e/Cargo.toml#L16

Reply:
Note that there is a `serde` feature on the crate to add serialization support:

```rust
let parsed = serde_json::from_str::<HashMap<usize, bool>>(r#"
    {
        "42": true
    }
"#);
println!("{:?}", parsed);
```

Reply:
So I am guessing from the context that `optional = true` is our ticket to having BigDecimal serialization, that this is the preferred approach and my solution is essentially a hack.

Reply:
Actually, I think the `Optional<BigDecimal>` wasn't the problem here (if that what the optional flag you are referring to is supposed to do). The problem is with `struct { field: BigDecimal }`. I will try your suggestion anyway, but I suspect this isn't a work around for my use of `#[serde(with = "serde_with::rust::display_fromstr")]` here.

Reply:
Awesome!

Reply:
So [cargo has the concept of optional features](https://doc.rust-lang.org/cargo/reference/features.html) to conditionally include additional code and functionality. This means it should be possible to add the `bigdecimal` dependency in a way that it includes `serde` serialization support. Specifically in the `Cargo.toml`:
```toml
bigdecimal = { version = "0.2", features = ["serde"] }
```